sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -215,8 +215,8 @@ class Analyzer(
     }
 
     def substituteCTE(plan: LogicalPlan, cteRelations: Seq[(String, LogicalPlan)]): LogicalPlan = {
-      plan resolveOperatorsDown {
-        case u: UnresolvedRelation =>
+      plan resolveOperatorsUp {
```

Comment:
Also, could you make a unit test for this case? Actually, for the current test case, line 219 looks enough to me.
> CTESubstitution transforms down, this means it will keep replacing t with itself, creating an infinite recursion.

Reply:
I am just being defensive here. I am not sure if there is a sane way of unit testing this. Let me roll this back.